tensorrt_llm/executor/worker.py

```diff
@@ -567,6 +567,12 @@ def shutdown(self):
             self.engine.shutdown()
             self.engine = None
 
+            if hasattr(
+                    self._executor_config, "checkpoint_loader"
+            ) and self._executor_config.checkpoint_loader is not None:
+                if hasattr(self._executor_config.checkpoint_loader, "cleanup"):
+                    self._executor_config.checkpoint_loader.cleanup()
```

Comment:
All `checkpoint_loader`s will have `def cleanup` since:
```python
@abstractmethod
def cleanup(self) -> None:
    raise NotImplementedError
```
is defined on the base. On the other hand - when you **call** cleanup in L574 and it isn't specified in the subclasses, shutdown will throw because of `NotImplementedError`. 
To fix this:
1. Either you call cleanup confidently without the hasattr check because you are confident all loaders will have implemented the abstract methods.
2. If some for reason you are afraid the `cleanup` won't *exist*, just:
```
try:
    self._executor_config.checkpoint_loader.cleanup()
except AttributeError:
    logger.warning(f"cleanup function not located on loader {loader}")
    pass
```
Or something.

Or am I misunderstanding something?